File: javascript/hamming/hamming.js
class Hamming{

	compute(strandA, strandB){
		let counter = 0;
		if(strandA.length != strandB.length){
			throw new  Error('left and right strands must be of equal length');
		}else{
			for(let i=0; i<strandA.length; i++){
				if(strandA[i]!= strandB[i]){
					counter+= 1;
				}
			}

		}
		return counter;
	}
}

export default Hamming;

File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strandA, strandB)
   counter = 0

   if strandA.length != strandB.length
    raise ArgumentError  
    
   end

  for i in 0..strandA.length do 
    if strandA[i] != strandB[i]
      counter += 1
    end
  end
  return counter
  end
end